rancher/charts
# Add network policy for RKE2 Ingress Nginx

## Issue: 

Addresses issue raised in https://github.com/rancher/rancher/issues/45603.

## Problem

Prometheus target failing to come up

## Solution

Using the modified chart in this PR enables the target to come up

## Testing


- Install rancher-monitoring-crd through this PR
- Install rancher-monitoring using following Helm command:
  ```sh
  helm install rancher-monitoring charts/rancher-monitoring/105.0.0-rc1+up57.0.3/ \
  --namespace=cattle-monitoring-system \
  --set alertmanager.alertmanagerSpec.configSecret=alertmanager-rancher-monitoring-alertmanager \
  --set alertmanager.alertmanagerSpec.useExistingSecret=true \
  --set global.cattle.clusterId=local \
  --set global.cattle.clusterName=local \
  --set global.cattle.rkePathPrefix="" \
  --set global.cattle.rkeWindowsPathPrefix="" \
  --set global.cattle.systemDefaultRegistry="" \
  --set global.cattle.systemProjectId=p-cwh9d \
  --set global.cattle.url=https://13.202.137.42.sslip.io \
  --set global.systemDefaultRegistry="" \
  --set prometheus.prometheusSpec.evaluationInterval=1m \
  --set prometheus.prometheusSpec.retentionSize=50GiB \
  --set prometheus.prometheusSpec.scrapeInterval=1m \
  --set rke2ControllerManager.enabled=true \
  --set rke2Etcd.enabled=true \
  --set rke2IngressNginx.enabled=true \
  --set rke2IngressNginx.networkPolicy.enabled=true \
  --set rke2Proxy.enabled=true \
  --set rke2Scheduler.enabled=true
  ````
- Go to Rancher UI; Monitoring; Prometheus Targets
- In the Prometheus UI search for the ServiceMonitor `serviceMonitor/kube-system/rancher-monitoring-ingress-nginx/0`. It should be `UP`.

## Engineering Testing
### Manual Testing


### Automated Testing


## QA Testing Considerations


### Regressions Considerations


## Backporting considerations
